cdap-ui/app/cdap/components/DataPrep/store/DataPrepActionCreator.js

```diff
@@ -296,3 +298,271 @@ export function setVisualizationState(state) {
+  }
+}
+
+export async function saveTargetDataModelStates() {
+  const params = {
+    context: NamespaceStore.getState().selectedNamespace,
+    workspaceId: DataPrepStore.getState().dataprep.workspaceId,
+  };
+
+  const { targetDataModel, targetModel } = DataPrepStore.getState().dataprep;
+  const newDataModelId = targetDataModel ? targetDataModel.id : null;
+  const newDataModelRevision = targetDataModel ? targetDataModel.revision : null;
+  const newModelId = targetModel ? targetModel.id : null;
+
+  // These properties were populated by MyDataPrepApi.getWorkspace API
+  const {
+    dataModel,
+    dataModelRevision,
+    dataModelModel,
+  } = DataPrepStore.getState().dataprep.properties;
+  const oldDataModelId = dataModel || null;
+  const oldDataModelRevision = isFinite(dataModelRevision) ? Number(dataModelRevision) : null;
+  const oldModelId = dataModelModel || null;
+
+  if (oldDataModelId !== newDataModelId || oldDataModelRevision !== newDataModelRevision) {
+    if (oldDataModelId !== null) {
+      await MyDataPrepApi.detachDataModel(params).toPromise();
+    }
+    if (newDataModelId !== null) {
+      await MyDataPrepApi.attachDataModel(params, {
+        id: newDataModelId,
+        revision: newDataModelRevision,
+      }).toPromise();
+    }
+  }
+
+  if (oldModelId !== newModelId) {
+    if (oldModelId !== null) {
+      await MyDataPrepApi.detachModel(
+        Object.assign(
+          {
+            modelId: oldModelId,
+          },
+          params
+        )
+      ).toPromise();
+    }
+    if (newModelId !== null) {
+      await MyDataPrepApi.attachModel(params, {
+        id: newModelId,
+      }).toPromise();
+    }
+  }
+
+  DataPrepStore.dispatch({
+    type: DataPrepActions.setProperties,
+    payload: {
+      properties: {
+        dataModel: newDataModelId,
+        dataModelRevision: newDataModelRevision,
+        dataModelModel: newModelId,
+      },
+    },
+  });
+}
+
+export async function fetchDataModelList(url) {
+  const namespace = NamespaceStore.getState().selectedNamespace;
+  const params = {
+    context: namespace,
+  };
+
+  await MyDataPrepApi.addDataModels(params, { url }).toPromise();
+  const response = await MyDataPrepApi.getDataModels(params).toPromise();
+  const dataModelList = response.values.map((dataModel) => ({
+    uuid: uuidV4(),
+    id: dataModel['namespacedId'].id,
+    revision: dataModel.revision,
+    name: dataModel.displayName,
+    description: dataModel.description,
+    url,
+  }));
+  dataModelList.sort((a, b) => a.name.localeCompare(b.name));
+
+  DataPrepStore.dispatch({
+    type: DataPrepActions.setDataModelList,
+    payload: {
+      dataModelList,
+    },
+  });
+
+  return dataModelList;
+}
+
+export async function fetchModelList(dataModel) {
+  const params = {
+    context: NamespaceStore.getState().selectedNamespace,
+    dataModelId: dataModel.id,
+    dataModelRevision: dataModel.revision,
+  };
+
+  // Response message example:
+  // {
```

Comment:
nit: Please make this multiline comment.
```
/**
 * example response
 */
```

Reply:
Done.